--- a/crates/wow-store/src/model.rs
+++ b/crates/wow-store/src/model.rs
@@ -445,6 +445,8 @@
 pub struct CatalogChange {
     catalog: CatalogName,
     path: CatalogPath,
+    #[serde(skip_serializing_if = "Option::is_none")]
     previous: Option<ObjectId>,
+    #[serde(skip_serializing_if = "Option::is_none")]
     current: Option<ObjectId>,
 }
@@ -544,5 +546,6 @@
     operation_id: OperationId,
     request_digest: RequestDigest,
     state: OperationState,
+    #[serde(skip_serializing_if = "Option::is_none")]
     result_object_id: Option<ObjectId>,
 }
